### ISSUE TITLE
NameError: name '_LOGGER' is not defined

### ISSUE THREAD

**Issue by atraunbauer** [databio/pepatac#266]
Hi, when doing a test run with the test1  files provided the script is running until the test1_bamQC.tsv should be created. At this point I am getting an error message:

NameError: name '_LOGGER' is not defined

I attached the PEPATAC_log.txt

Could you please help to solve this? Thank you so much!

[PEPATAC_log.txt](https://github.com/databio/pepatac/files/13964956/PEPATAC_log.txt)

**Comment by donaldcampbelljr** [comment]
Hello,

Just curious, did you use Looper to run your samples? https://github.com/pepkit/looper

I'm wondering if not using Looper would cause this to happen as Looper gets Logger.

**Comment by atraunbauer** [comment]
Hi,

thanks for getting back to me!

Yes, I used looper (version 1.6.0)  to run the test1 samples.

**Comment by donaldcampbelljr** [comment]
Did you happen to edit the file `bamQC.py`?
Can you confirm that this code is still present? 
https://github.com/databio/pepatac/blob/ffa7cc23ff9b11483f1ab317670d06bb886d6288/tools/bamQC.py#L243-L257

Specifically: `_LOGGER = logmuse.logger_via_cli(args, make_root=True)`

**Comment by atraunbauer** [comment]
Yes _LOGGER = logmuse.logger_via_cli(args, make_root=True) is still there. I didn't edit the bamQC.py file.

# parallel processed computation of matrix for each chromosome
if __name__ == "__main__":

    args = parse_args(sys.argv[1:])
    _LOGGER = logmuse.logger_via_cli(args, make_root=True)

    qc = bamQC(reads_filename=args.infile,
               out_filename=args.outfile,
               n_proc=args.cores,
               verbosity=args.verbosity)

    qc.register_files()
    good_chromosomes = qc.run()

    _LOGGER.debug("Reduce step (merge files)...")
    qc.combine(good_chromosomes)

**Comment by jpsmith5** [comment]
Confirmed: `_LOGGER` was defined only inside the `if __name__ == "__main__":` block of `tools/bamQC.py`, so when `pararead` workers re-import the module under multiprocessing `spawn` (default on macOS, and used elsewhere when fork is unavailable), the class methods hit `NameError`. Fixed on `dev` by adding a module-level `_LOGGER = logging.getLogger(__name__)` fallback above the class definition; the `__main__` block still upgrades it to a logmuse-configured logger for the parent process. Same fix applied to `tools/bamSitesToWig.py` defensively (the active pipeline replaced it with `gtars uniwig` in 0.12.0, but the script is still in the repo). Ships in upcoming v0.14.0. Closing — the `spawn` failure mode is hard to reproduce on Linux test runners (which default to fork), so macOS users please report back if you still hit a NameError against the new release.
